File: csharp/Agent/BitoAgent.cs
using System;
using System.Threading.Tasks;

namespace Maestro.Agent
{
    public class BitoAgent
    {
        public string ModelProfile { get; private set; } = "ADVANCED";
        public int MaxContext { get; private set; } = 240000;

        public void SetModelProfile(string profile)
        {
            if (profile == "BASIC") {
                MaxContext = 40000;
            } else {
                MaxContext = 240000;
            }
            ModelProfile = profile;
            Console.WriteLine($"Bito Model Profile set to: {ModelProfile} (Limit: {MaxContext})");
        }

        public async Task<string> InjectPromptMacroAsync(string template, string fileContent)
        {
            await Task.Delay(50);

            string result = template.Replace("{{%input%}}", fileContent);

            if (result.Length > MaxContext) {
                throw new Exception($"Injected prompt exceeds maximum context length of {MaxContext}");
            }

            return result;
        }
    }
}

File: java/src/main/java/com/maestro/agent/BitoAgent.java
package com.maestro.agent;

import java.util.concurrent.CompletableFuture;

public class BitoAgent {
    private String modelProfile = "ADVANCED";
    private int maxContext = 240000;

    public void setModelProfile(String profile) {
        if ("BASIC".equals(profile)) {
            this.maxContext = 40000;
        } else {
            this.maxContext = 240000;
        }
        this.modelProfile = profile;
        System.out.println("Bito Model Profile set to: " + this.modelProfile + " (Limit: " + this.maxContext + ")");
    }

    public CompletableFuture<String> injectPromptMacro(String template, String fileContent) {
        return CompletableFuture.supplyAsync(() -> {
            try { Thread.sleep(50); } catch (InterruptedException e) {}

            String result = template.replace("{{%input%}}", fileContent);

            if (result.length() > this.maxContext) {
                throw new RuntimeException("Injected prompt exceeds maximum context length of " + this.maxContext);
            }

            return result;
        });
    }
}
